lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -300,9 +300,17 @@ static CodingKeysValidity hasValidCodingKeysEnum(TypeChecker &tc,
   if (!tc.conformsToProtocol(codingKeysType, codingKeyProto,
                              target->getDeclContext(),
                              ConformanceCheckFlags::Used)) {
-    tc.diagnose(codingKeysTypeDecl->getLoc(),
-                diag::codable_codingkeys_type_does_not_conform_here,
+    // If CodingKeys is a typealias which doesn't point to a valid nominal type,
+    // codingKeysTypeDecl will be nullptr here. In that case, we need to warn on
+    // the location of the usage, since there isn't an underlying type to
+    // diagnose on.
+    SourceLoc loc = codingKeysTypeDecl ?
+                    codingKeysTypeDecl->getLoc() :
+                    cast<TypeDecl>(result)->getLoc();
+
+    tc.diagnose(loc, diag::codable_codingkeys_type_does_not_conform_here,
```

Comment:
Perhaps you can suppress this diagnostic if `codingKeysTypeDecl->isInvalid()`. In your test case you're emitting it twice which is not ideal

Reply:
If you conform to `Codable`, we go through synthesis twice — once for `Decodable`, and once for `Encodable`. There isn't a great way to signal that `Decodable` synthesis has gone through so `Encodable` synthesis shouldn't warn, and vice versa.

We can also suppress the diagnostic here if `codingKeysTypeDecl` is `nullptr` and just let the error message through. I do think it can be helpful to have the note as well to indicate that we at least looked at this decl, but it's not the end of the world either way.